pkg/gatherers/clusterconfig/node_logs.go

```diff
@@ -0,0 +1,149 @@
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatherNodeLogs fetches the node logs from journal unit
+//
+// Response see https://docs.openshift.com/container-platform/4.8/rest_api/node_apis/node-core-v1.html#apiv1nodesnameproxypath
+//
+// * Location in archive: config/nodes/logs/
+// * See: docs/insights-archive-sample/config/nodes/logs
+// * Id in config: node_logs
+func (g *Gatherer) GatherNodeLogs(ctx context.Context) ([]record.Record, []error) {
+	clientSet, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+	return gatherNodeLogs(ctx, clientSet.CoreV1())
+}
+
+func gatherNodeLogs(ctx context.Context, client corev1client.CoreV1Interface) ([]record.Record, []error) {
+	nodes, err := client.Nodes().List(ctx, metav1.ListOptions{LabelSelector: "node-role.kubernetes.io/master"})
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	records, errs := nodeLogRecords(client.RESTClient(), nodes)
+	return records, errs
+}
+
+// nodeLogRecords generate the records and errors list
+func nodeLogRecords(restClient rest.Interface, nodes *corev1.NodeList) ([]record.Record, []error) {
+	var errs []error
+	records := make([]record.Record, 0)
+
+	bufferSize := recorder.MaxArchiveSize * logCompressionRatio / len(nodes.Items) / 2
+	klog.V(2).Infof("Maximum buffer size: %v bytes", bufferSize)
+	buf := bytes.NewBuffer(make([]byte, 0, bufferSize))
+
+	for i := range nodes.Items {
+		name := nodes.Items[i].Name
+		uri := nodeLogResourceURI(restClient, name)
+		req := requestNodeLog(restClient, uri, logNodeMaxTailLines, logNodeUnit)
+
+		logString, err := nodeLogString(req, buf, bufferSize)
+		if err != nil {
+			klog.V(2).Infof("Error: %q", err)
+			errs = append(errs, err)
+		}
+
+		records = append(records, record.Record{
+			Name: fmt.Sprintf("config/node/logs/%s.log", name),
+			Item: marshal.Raw{Str: logString},
+		})
+	}
+
+	return records, errs
+}
+
+// nodeLogResourceURI creates the resource path URI to be fetched
+func nodeLogResourceURI(client rest.Interface, name string) string {
+	return client.Get().
+		Name(name).
+		Resource("nodes").SubResource("proxy", "logs").
+		Suffix("journal").URL().Path
+}
+
+// requestNodeLog creates the request to the API to retrieve the resource stream
+func requestNodeLog(client rest.Interface, uri string, tail int, unit string) *rest.Request {
+	return client.Get().RequestURI(uri).
+		SetHeader("Accept", "text/plain, */*").
+		SetHeader("Accept-Encoding", "gzip").
+		Param("tail", strconv.Itoa(tail)).
+		Param("unit", unit)
+}
+
+// nodeLogString retrieve the data from the stream, decompress it (if necessary) and return the string
+func nodeLogString(req *rest.Request, out *bytes.Buffer, size int) (string, error) {
+	in, err := req.Stream(context.TODO())
```

Comment:
I think we should not use `context.TODO()`, but rather use the context passed to the gatherer.